### ISSUE TITLE
Parsing error: Unexpected token[s] 

### ISSUE THREAD

**Issue by FQ400** [eslint/eslint#8857]
**Tell us about your environment**

* **ESLint Version:** 4.1.1
* **Node Version:** 8.1.2
* **npm Version:** 5.0.4

**What parser (default, Babel-ESLint, etc.) are you using?**
I think `babel-eslint`. The package is at least installed in the project. The command I use to run eslint is `eslint --ext .jsx --ext .js src/ tests/`. How can I be sure?

**Please show your full configuration:**

```
{
  "env": {
    "browser": true,
    "es6": true,
    "jest": true
  },
  "extends": [
    "eslint:recommended",
    "plugin:inferno/recommended",
    "holidaycheck/es2015"
  ],
  "parserOptions": {
    "ecmaFeatures": {
      "jsx": true
    },
    "sourceType": "module"
  },
  "plugins": [
    "compat",
    "inferno"
  ],
  "globals": {
    "TestSupport": true,
    "global": true
  },
  "rules": {
    "indent": [
      "error",
      2
    ],
    "linebreak-style": [
      "error",
      "unix"
    ],
    "quotes": [
      "error",
      "single"
    ],
    "semi": [
      "error",
      "always"
    ],
    "eol-last": [
      "warn",
      "always"
    ],
    "compat/compat": 2,
    "inferno/inferno-in-jsx-scope": 0,
    "no-extra-parens": [
      "error",
      "all",
      { "ignoreJSX": "all" }
    ],
    "no-shadow": ["warn", { "builtinGlobals": true, "hoist": "functions", "allow": [] }],
    "consistent-this": 0,
    "max-statements": 1,
    "no-warning-comments": 1,
    "handle-callback-err": 0,
    "camelcase": 1
  }
}
```

**What did you do? Please include the actual source code causing the issue.**

I updated eslint from `4.0.0` to `4.1.1` without code changes.

Example 1:
```js
const defaultChildProps = {
  ...defaultProps   // Parsing Error in this line
};
const renderChildComponent = (props = {}) => renderIntoDocument(<SelectInput {...defaultChildProps} {...props} />);
```

Example 2:
```js
  test('throws on status code 401 - Unauthorized', async () => {    // Parsing Error in this line
    const statusText = 'Unauthorized';
    const fetchStub = createFetchSandbox({ status: 401, statusText });

    await expect(getSignedRequest(requestPayload, fetchStub)).rejects.toMatchObject(new Error(statusText));
  });

```

**What did you expect to happen?**

I expected no parsing errors.

**What actually happened? Please include the actual, raw output from ESLint.**

For example 1 it outputs `error  Parsing error: Unexpected token ..`
For example 2 it outputs `error  Parsing error: Unexpected token =>`

`// Parsing Error in this line` was added from me for understanding purpose.

**Comment by ilyavolodin** [comment]
Based on your configuration, you are using default parser (espree). You are getting those errors, because you haven't specified `ecmaVersion: 2017` under `parserOptions` (http://eslint.org/docs/user-guide/configuring#specifying-parser-options). In order to get the first example to work, you also have to specify `experimentalObjectRestSpread: true` property under `ecmaFeatures`.

**Comment by FQ400** [comment]
Thank you for the quick response. Your advice fixed my issue. No further question. 😄
